### ISSUE TITLE
Open question - Winium.Elements.Desktop methods are not being used when remote webdriver is utilitzed

### ISSUE THREAD

**Issue by muruga85** [2gis/Winium.Desktop#117]
I'm currently using RemoteWebDriver to run the winium scripts in a remote server but unfortunately none of the Winium.Elements.Desktop methods are working? Can you please show some suggestions on this?

 var dc = new DesiredCapabilities();
 dc.SetCapability("app", _applicationExePath);
 _driver = new RemoteWebDriver(new Uri(string.Format("http://{0}:9999/",server)), dc);

Code which doesnt work:

 Winium.Elements.Desktop.ComboBox combobox = new Winium.Elements.Desktop.ComboBox(element);
 combobox.Expand();

**Comment by NickAb** [comment]
Please provide driver logs. Make sure that the `element` is exactly of type ComboBox in your app.

You can also try running tests from https://github.com/2gis/Winium.Desktop/blob/master/src/TestApps.Tests/WpfTestApplication.Tests/WiniumCommandTests/ComboBoxCommandsTests.cs to see if it breaks on our test app.

**Comment by muruga85** [comment]
Yes, the element is exactly of type ComboBox in my app. Moreover, if I run the same code using a normal driver instead of RemoteWebDriver through continous integration tool Jenkins, it works. Below is the code for that:

DesktopOptions options = new DesktopOptions { ApplicationPath = _applicationExePath };
 _driver = new WiniumDriver(service, options);
return _driver

If we invoke the driver this way, the ComboBox works. Please advise as we are in the verge of completing Winium POC for our project.

**Comment by muruga85** [comment]
Below is the exception I receive:

An exception of type 'System.NullReferenceException' occurred in WebDriver.dll but was not handled in user code

Additional information: Object reference not set to an instance of an object.

**Comment by NickAb** [comment]
`WiniumDriver` extends `RemoteWebDriver`, so there is no need to use `RemoteWebDriver` directly. Just use `WiniumDriver` both on Jenkins and locally.

**Comment by muruga85** [comment]
How can we use WiniumDriver instead of RemoteWebDriver in case if I want to run the winium scripts on a remote system? I mean, how can we supply remote system IP as a parameter to WiniumDriver like we do for RemoteWebDriver. Expecting your response at the earliest.

**Comment by NickAb** [comment]
You can use
```cs
public WiniumDriver(Uri remoteAddress, IWiniumOptions options)
```
See https://github.com/2gis/Winium/blob/master/dotnet/src/OpenQA.Selenium.Winium/WiniumDriver.cs#L109 for all options on how you can construct WiniumDriver.

**Comment by muruga85** [comment]
This actually doesnt work for me. Please see below:

 DesktopOptions options = new DesktopOptions { ApplicationPath = _applicationExePath };
 _driver = new WiniumDriver((new Uri(string.Format("http://{0}:9999/", server)), options);

Error displayed upon building:

The best overloaded method match for 'OpenQA.Selenium.Winium.WiniumDriver.WiniumDriver(OpenQA.Selenium.Winium.WiniumDriverService, OpenQA.Selenium.Winium.IWiniumOptions)' has some invalid arguments

Argument 1: cannot convert from 'System.Uri' to 'OpenQA.Selenium.Winium.WiniumDriverService'

I've already moved winium.desktop.driver to the remote server using PSEXESVC.exe and the driver is already started in the remote server where the application is installed. Kindly advise on how to proceed now

**Comment by muruga85** [comment]
Also, when I viewed the metadata, I dont find a constructor for URI. Please see below:

using OpenQA.Selenium.Remote;
using System;

namespace OpenQA.Selenium.Winium
{
    public class WiniumDriver : RemoteWebDriver
    {
        public WiniumDriver(string winiumDriverDirectory, IWiniumOptions options);
        public WiniumDriver(WiniumDriverService service, IWiniumOptions options);
        public WiniumDriver(string winiumDriverDirectory, IWiniumOptions options, TimeSpan commandTimeout);
        public WiniumDriver(WiniumDriverService service, IWiniumOptions options, TimeSpan commandTimeout);
    }
}

**Comment by NickAb** [comment]
Are you using 0.1.0 of OpenQA.Selenium.Winium? Could you please download latest version of https://github.com/2gis/Winium source code and build it. It does include constructor that takes remote server uri as parameter.

@skyline-gleb, can we release a new version to nuget?

**Comment by muruga85** [comment]
Yes, I'm using 0.1.0 version of Winium.WebDriver. I did download the latest version of Winium source code and built it. It did have the constructor which takes uri as a parameter. How can I reference that until we get the latest version to nuget? Any quick response is much appreciated

**Comment by muruga85** [comment]
I did build the latest source code from github, built it and reference that dll to my project after removing the existing Winium.WebDriver.Dll reference from nuget. I'm now seeing the constructor that takes uri as parameter. Hoping to have it from nuget quite soon. Thanks

**Comment by NickAb** [comment]
You can build Winium project in Release configuration. And copy resulting dlls into your project folder and commit. You can add reference to local dll files by right clicking `References` and selecting `Add Reference... → Assemblies → Browse...`

**Comment by muruga85** [comment]
Yes, as I mentioned earlier I did manually reference those DLL's. Let's also wait for the latest version in nuget.

**Comment by skyline-gleb** [comment]
@muruga85 hi. I released 0.1.1 with remote address constructor, please check this. Thank you!

**Comment by muruga85** [comment]
Thanks @skyline-gleb and @NickAb for getting this done. Expecting the same support in the future.
